app/uk/gov/hmrc/helptosave/actors/EligibilityStatsActor.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosave.actors
+
+import java.time.LocalTime
+import java.util.concurrent.TimeUnit
+
+import akka.actor.{Actor, Cancellable, Props, Scheduler}
+import akka.pattern.pipe
+import com.typesafe.config.Config
+import uk.gov.hmrc.helptosave.actors.EligibilityStatsActor.{GetStats, GetStatsResponse}
+import uk.gov.hmrc.helptosave.services.EligibilityStatsService
+import uk.gov.hmrc.helptosave.util.{Logging, Time}
+
+import scala.concurrent.duration.Duration
+
+class EligibilityStatsActor(scheduler:               Scheduler,
+                            config:                  Config,
+                            timeCalculator:          TimeCalculator,
+                            eligibilityStatsService: EligibilityStatsService,
+                            eligibilityStatsHandler: EligibilityStatsHandler) extends Actor with Logging {
+
+  import context.dispatcher
+
+  var eligibilityStatsJob: Option[Cancellable] = None
+
+  override def receive: Receive = {
+    case GetStats ⇒
+      logger.info("Getting eligibility stats from mongo")
+      eligibilityStatsService.getEligibilityStats().map(GetStatsResponse) pipeTo self
+
+    case r: GetStatsResponse ⇒
+      eligibilityStatsHandler.handleStats(r.result)
+  }
+
+  def scheduleStats(): Cancellable = {
+    val scheduleStart = LocalTime.parse(config.getString("eligibility-stats.trigger-time"))
+    val timeUntilNextTrigger = timeCalculator.timeUntil(scheduleStart)
+    logger.info(s"Scheduling eligibility stats job in ${Time.nanosToPrettyString(timeUntilNextTrigger.toNanos)}")
+    scheduler.schedule(timeUntilNextTrigger, Duration(24, TimeUnit.HOURS), self, GetStats)
```

Comment:
job runs everyday at 1.00 AM

Reply:
it'd be nice to be able to run it perhaps at a more regular frequency - perhaps it'd be more flexible here to put in a list of times or an interval between jobs?

Reply:
do we want to run this more than once a day ?  because it scans all mongo records in the collection, which can be a performance hit because if there are 3 instances in production there will be 3 such jobs at the moment, all running at the same time which is over doing for the same report actually...

Reply:
you've got a point there - we should probably be using the lock stuff to make sure that only one instance of this job is running per environment. We already have that lock stuff in the proxy so you should be able to re-use it. Having that in place, it'd be nice to perhaps log this information every 1 or 2 hours to be able to measure rates of enrolments